source/src/BNFC/Backend/Haskell/CFtoAlex3.hs

```diff
@@ -188,33 +188,49 @@ restOfAlex tokenText cf = concat
     , ""
     , "-- | Get the position of a token."
     , "tokenPosn :: Token -> Posn"
-    , "tokenPosn (PT p _) = p"
-    , "tokenPosn (Err p)  = p"
+    , "tokenPosn (PT posn _len _tok) = posn"
+    , "tokenPosn (Err posn)  = posn"
     , ""
-    , "-- | Get line and column of a token."
+    , "-- | Get the length of a token."
+    , "tokenLen :: Token -> Int"
+    , "tokenLen (PT _posn len _tok) = len"
+    , "tokenLen (Err _)  = 0"
+    , ""
+    , "-- | Get start line and column of a token."
     , "tokenLineCol :: Token -> (Int, Int)"
     , "tokenLineCol = posLineCol . tokenPosn"
     , ""
+    , "-- | Get end line and column of a token."
+    , "tokenLineColEnd :: Token -> (Int, Int)"
+    , "tokenLineColEnd t = (l, c + n)"
```

Comment:
What if a token spans several lines?  Note that the user can define their own tokens that may contain newline characters.